adapters/insticator/insticator.go

```diff
@@ -86,20 +86,22 @@ func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapte
 
 	// Create a copy of the request to avoid modifying the original request
 	requestCopy := *request
+	isPublisherIdPopulated := false // Flag to track if populatePublisherId has been called
 
 	for i := 0; i < len(request.Imp); i++ {
-		impCopy, impKey, err := makeImps(request.Imp[i])
+		impCopy, impKey, publisherId, err := makeImps(request.Imp[i])
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
 
-		// Populate site.publisher.id from imp extension
-		if requestCopy.Site != nil || requestCopy.App != nil {
-			if err := populatePublisherId(&impCopy, &requestCopy); err != nil {
+		// Populate publisher.id from imp extension
+		if !isPublisherIdPopulated && (requestCopy.Site != nil || requestCopy.App != nil) {
```

Comment:
You can simplify this to just `if !isPublisherIdPopulated {` since you've only declared support for site and app (not dooh) in your YAML file which means your adapter will only be called if site or app is present.